src/auth/auth-api-request.ts

```diff
@@ -725,6 +794,44 @@ export abstract class AbstractAuthRequestHandler {
     return (validator.isNonNullObject(response) && response.error && response.error.message) || null;
   }
 
+  private static addUidToRequest(id: UidIdentifier, request: GetAccountInfoRequest): GetAccountInfoRequest {
+    if (!validator.isUid(id.uid)) {
+      throw new FirebaseAuthError(AuthClientErrorCode.INVALID_UID);
+    }
+    request.localId ? request.localId.push(id.uid) : request.localId = [id.uid];
+    return request;
+  }
+
+  private static addEmailToRequest(id: EmailIdentifier, request: GetAccountInfoRequest): GetAccountInfoRequest {
+    if (!validator.isEmail(id.email)) {
+      throw new FirebaseAuthError(AuthClientErrorCode.INVALID_EMAIL);
+    }
+    request.email ? request.email.push(id.email) : request.email = [id.email];
+    return request;
+  }
+
+  private static addPhoneToRequest(id: PhoneIdentifier, request: GetAccountInfoRequest): GetAccountInfoRequest {
+    if (!validator.isPhoneNumber(id.phoneNumber)) {
+      throw new FirebaseAuthError(AuthClientErrorCode.INVALID_PHONE_NUMBER);
+    }
+    request.phoneNumber ? request.phoneNumber.push(id.phoneNumber) : request.phoneNumber = [id.phoneNumber];
+    return request;
+  }
+
+  private static addProviderToRequest(id: ProviderIdentifier, request: GetAccountInfoRequest): GetAccountInfoRequest {
+    if (!validator.isNonEmptyString(id.providerUid) || !validator.isNonEmptyString(id.providerId)) {
+      throw new FirebaseAuthError(AuthClientErrorCode.INVALID_PROVIDER_ID);
```

Comment:
I think invalid provider ID is different that invalid provider uid. The error is confusing here.